code/game/machinery/mind_machine.dm

```diff
@@ -0,0 +1,856 @@
+/obj/machinery/mind_machine
+	name = "\improper Mind Machine"
+	icon = 'icons/obj/mind_machine.dmi'
+	density = 1
+	anchored = 1
+	use_power = 1
+	idle_power_usage = 50
+	active_power_usage = 2000
+	light_power_on = 1
+	machine_flags = SCREWTOGGLE | CROWDESTROY | EMAGGABLE | FIXED2WORK | WRENCHMOVE
+
+#define CURRENTLY_ACTIVE "Currently active: please wait"
+#define PODS_UNLOCKED "Pods already unlocked"
+#define PODS_LOCKED "Pods are locked"
+#define REQ_SCAN "Require scan"
+#define DELAY_CAUTION "Proceed with caution: supervisor advised"
+#define MISSING_OCC "Missing occupant"
+#define CONDUIT_CHARGE "Conduit requires charge"
+#define SHIELD_ERROR "Critical error, aborting"
+#define LIVING_REQUIRED "Living mind required"
+#define SHARD_REACTION "Warning: Unknown internal reaction"
+#define SCANNER_INSUFF "Scanners insufficient"
+#define UNKNOWN_INT "Unknown interruption"
+#define SUM_TOO_LOW "Sum intelligence too low"
+#define DELIVERY_ERROR "Error in mind delivery"
+#define MIND_DAMAGED "Mind sustained damage"
+#define SAFETY_CONSUMED "Safety consumed"
+#define TRANSPORT_ANOM "Mind transport anomaly"
+#define NOERROR "No errors"
+
+#define LOWER "Lower"
+#define HIGHER "Higher"
+#define SILICON "Artificial"
+#define SHIELDED "Shielded"
+
+/obj/machinery/mind_machine/mind_machine_hub
+	name = "\improper Mind Machine Hub"
+	icon_state = "mind_hub"
+	desc = "The main hub of a complete mind machine setup. Placed between two mind pods and used to control and manage the transfer. Houses an experimental bluespace conduit which uses bluespace crystals for charge."
+	var/bluespaceConduit = 0 //Swap ammo
+	var/soulShardSafety = FALSE	//Day ruin insurance
+	var/obj/machinery/mind_machine/mind_machine_pod/connectOne //All "variableNameOne" refer to information about the connectOne pod
+	var/obj/machinery/mind_machine/mind_machine_pod/connectTwo //Likewise for variableNameTwo for connectTwo
+	var/podsConnected = FALSE
+	var/mob/living/occupantOne = null
+	var/mob/living/occupantTwo = null
+	var/occupantStatOne = null
+	var/occupantStatTwo = null
+	var/mindTypeOne = null	//Player mind, simple mob, silicon, etc
+	var/mindTypeTwo = null
```

Comment:
You really don't need these vars since the occupants are typecast.

Reply:
Those ones I do. I got rid of name for that reason but stat serves a single minor purpose in allowing people who were scanned when alive to still be swapped after death. I like that as a feature and want to keep it. It also admittedly just makes it much easier for me to have the UI display a their actual state rather than a number.